From dd77934ca6e2e8b2b894f2460017323b33b619b6 Mon Sep 17 00:00:00 2001
From: matoro <matoro@users.noreply.github.com>
Date: Thu, 23 May 2024 16:10:32 -0400
Subject: [PATCH] [ruby/readline-ext] Skip test_interrupt_in_other_thread on
 arm32-linux

This is a combination of main Ruby commit
https://github.com/ruby/ruby/commit/aefc98891c42024039f19ef45bdfe93fbc590b7c
and my PR correcting the regex https://github.com/ruby/ruby/pull/10819.
Upstream Ruby requests that changes to this test go to readline-ext repo
before being backported to 3.2 branch.

https://github.com/ruby/readline-ext/commit/868f873a78
---
 test/readline/test_readline.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/readline/test_readline.rb b/test/readline/test_readline.rb
index 7838f4f52e864e..ab70e4371122f0 100644
--- a/test/readline/test_readline.rb
+++ b/test/readline/test_readline.rb
@@ -496,8 +496,8 @@ def test_interrupt_in_other_thread
     # Maybe the same issue: https://github.com/facebookresearch/nle/issues/120
     omit if /i[3-6]86-linux/ =~ RUBY_PLATFORM
 
-    # Skip arm32-linux (Travis CI).
-    omit "Skip arm32-linux" if /armv.+l-linux/ =~ RUBY_PLATFORM
+    # Skip arm32-linux (Travis CI).  See aefc988 in main ruby repo.
+    omit "Skip arm32-linux" if /armv[0-9+][a-z]-linux/ =~ RUBY_PLATFORM
 
     if defined?(TestReadline) && self.class == TestReadline
       use = "use_ext_readline"
